kedro/io/tensorflow_model_data_set.py

```diff
@@ -0,0 +1,237 @@
+        >>> loaded_model = data_set.load()
+        >>> new_predictions = loaded_model.predict([...])
+        >>> np.testing.assert_allclose(predictions, new_predictions, rtol=1e-6, atol=1e-6)
+
+    """
+
+    DEFAULT_LOAD_ARGS = dict(custom_objects=None, compile=True)  # type: Dict[str, Any]
+
+    DEFAULT_SAVE_ARGS = dict(
+        overwrite=True,
+        include_optimizer=True,
+        save_format="tf",
+        signatures=None,
+        options=None,
+    )  # type: Dict[str, Any]
+
+    def __init__(
+        self,
+        filepath: str,
+        load_args: Dict[str, Any] = None,
+        save_args: Dict[str, Any] = None,
+    ) -> None:
+        """
+        Args:
+            filepath: Filepath to a TensorFlow model directory prefixed with a protocol like `s3://`.
+                If prefix is not provided `file` protocol (local filesystem) will be used.
+                The prefix should be any protocol supported by ``fsspec``.
+                Note: `http(s)` doesn't support versioning.
+            load_args: TensorFlow options for loading models.
+                custom_objects: Optional dictionary mapping names (strings) to custom classes or functions to be
+                considered during deserialization.
+                compile: Boolean, whether to compile the model after loading.
+                Here you can find all available arguments:
+                https://www.tensorflow.org/api_docs/python/tf/keras/models/load_model
+                All defaults are preserved.
+            save_args: TensorFlow options for saving models.
+                Here you can find all available arguments:
+                https://www.tensorflow.org/api_docs/python/tf/keras/models/save_model
+                All defaults are preserved, except for "index", which is set to False.
+                In TF this defaults to 'tf' in TF 2.X, and 'h5' in TF 1.X.
+        """
+        super().__init__()
+
+        self._filepath = filepath
+        self._data = None
+
+        # Handle default load and save arguments
+        self._load_args = copy.deepcopy(self.DEFAULT_LOAD_ARGS)
+        if load_args is not None:
+            self._load_args.update(load_args)
+        self._save_args = copy.deepcopy(self.DEFAULT_SAVE_ARGS)
+        if save_args is not None:
+            self._save_args.update(save_args)
+
+    def _load(self) -> None:
+        return tf.keras.models.load_model(self._filepath, **self._load_args)
+
+    def _save(self, data: tf.keras.Model) -> None:
+        tf.keras.models.save_model(data, self._filepath, **self._save_args)
+
+    def _exists(self) -> bool:
+        return Path(self._filepath).is_dir()
+
+    def _describe(self) -> Dict[str, Any]:
+        # todo(aleks) - could also expose model.summary()
+        return dict(
+            filepath=self._filepath,
+            load_args=self._load_args,
+            save_args=self._save_args,
+        )
+
+
+class TensorFlowModelVersionedDataset(AbstractVersionedDataSet):
```

Comment:
This doesn't need to be a separate class, you should only have one `TenorFlowModelDataSet` that inherits from `AbstractVersionedDataset` (which itself inherits from `AbstractDataSet`). Then users can turn versioning on/off with the `versioned` flag in `catalog.yml`.

Reply:
yep thats clear.

for own knowledge where is catalog.yml? assume its like a user config after you've setup a kedro workflow?

Reply:
Yep, once you generate your kedro project with `kedro new`, the catalog can be found in `conf/base/catalog.yml` which is where users specify their datasets.